snakemake/shell.py

```diff
@@ -134,7 +147,7 @@ def __new__(
             stdout=stdout,
             universal_newlines=iterable or None,
             close_fds=close_fds,
-            **cls._process_args
+            **cls._process_args,
```

Comment:
Please do the following for setting the variables in a portable way:
```
envvars = dict(os.environ)
envvars["OMP_NUM_THREADS"] = threads
...
```
and then pass that as `env=` argument to `sp.Popen`.

Reply:
@johanneskoester This is finally passing. Spent an over an hour figuring out that Popen insists on all values in the `env=` `dict` being of type `str`, so `threads has to be `str`...